src/main/java/net/blay09/mods/waystones/item/ItemBoundScroll.java

```diff
@@ -105,18 +105,32 @@ public EnumActionResult onItemUseFirst(EntityPlayer player, World world, BlockPo
     }
 
     @Override
-    public ItemStack onItemUseFinish(ItemStack itemStack, World world, EntityLivingBase entity) {
-        if (!world.isRemote && entity instanceof EntityPlayer) {
-            WaystoneEntry boundTo = getBoundTo((EntityPlayer) entity, itemStack);
-            if (boundTo != null) {
+    public ItemStack onItemUseFinish(ItemStack itemStack, World world, EntityLivingBase entity)
+    {
+        if (!world.isRemote && entity instanceof EntityPlayer)
+        {
+            EntityPlayer player = (EntityPlayer) entity;
+
+            WaystoneEntry boundTo = getBoundTo(player, itemStack);
+            if (boundTo != null)
+            {
                 double distance = entity.getDistance(boundTo.getPos().getX(), boundTo.getPos().getY(), boundTo.getPos().getZ());
-                if (distance <= 2.0) {
+                if (distance <= 2)
+                {
                     return itemStack;
                 }
 
-                if (WaystoneManager.teleportToWaystone((EntityPlayer) entity, boundTo)) {
-                    if (!((EntityPlayer) entity).capabilities.isCreativeMode) {
-                        itemStack.shrink(1);
+                TileWaystone tileWaystone = WaystoneManager.getWaystoneInWorld(boundTo);
+                if (tileWaystone != null) //If it's going to fail, it should fail here.  If it fails inside this block, I (Laike Endaril) messed up Blay's mod :P
+                {
+                    if (!player.capabilities.isCreativeMode) itemStack.shrink(1);
```

Comment:
This should stay inside the `teleportToWaystone`-if-block so that just in case the teleport does unexpectedly fail, the item won't be used up.

Reply:
Fair enough. I've also changed the code style back to what should match the rest of your project (I must've forgotten to do so after editing the first time).